rfcs/0007-two-phase-approx-count-distinct.md

```diff
@@ -0,0 +1,36 @@
+---
+feature: two_phase_approx_count_distinct
+authors:
+  - "soundOfDestiny"
+start_date: "2022/10/28"
+---
+
+# 2-phase Approx Count Distinct Agg
+
+## Summary
+
+2-phase Approx Count Distinct Agg which uses the bucket_id as distribution key to calculate vnode for 2-phase agg, as jon-chuang proposed in https://github.com/risingwavelabs/risingwave/issues/3414.
+
+## Motivation
+
+Currently the `partial_to_total_agg` of `Approx Count Distinct Agg` is `Sum0`, but it is a bad estimation because $Cardinality(S1 \cup S2) \neq Cardinality(S1) + Cardinality(S2)$.
+
+## Design
+
+* Alter group-by key of input into `bucket_id || group_by_keys`. (For example, in GlobalSimpleAgg whose `group_by_keys` is `None`, the new group-by key will be `bucket_id`.) We need to leverage `StreamExchange` to alter distribution key accordingly if necessary.
```

Comment:
will we introduce a new expression to evaluate the bucket_id? It sounds a little strange to me because no other things need that expression